[iOS] Adjust several API tests so that they pass when run against any iOS simulator

whsieh opened the pull request:
#### 874dbd53a389b72df8ec35cbe5b478f4d5327da5
<pre>
[iOS] Adjust several API tests so that they pass when run against any iOS simulator
<a href="https://bugs.webkit.org/show_bug.cgi?id=257246">https://bugs.webkit.org/show_bug.cgi?id=257246</a>
rdar://109743641

Reviewed by Aditya Keerthi.

Make some additional adjustments to several API tests, which currently pass in the iPhone SE
simulator but fail when run against other simulator models. These tests are:

• A subset of `WKWebView.SnapshotImage*`
• A subset of `AnimatedResize.*`
• `WebKitLegacy.RenderInContextSnapshot`

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Drive-by fix: address a (benign) build warning following 264080@main, where the definition of
`-runFromElement:toElement` is compiled twice, due to the fact that `DragAndDropSimulator.mm`
appears in _both_ the unified sources list as well as the Xcode project as a compilation target.
Address this by removing it as a compilation target in Xcode.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/AnimatedResize.mm:

Fix these tests by subclassing `TestWKWebView`, which normalizes safe area inset values between
different device models.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewSnapshot.mm:

Fix these tests by adopting `TestWKWebView` in place of `WKWebView`.

* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/SnapshotViaRenderInContext.mm:
(-[UIWebViewWithoutSafeArea _safeAreaInsetsForFrame:inSuperview:]):

Fix this test by creating a `UIWebView` subclass that simulates 0 safe area insets.

* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView _safeAreaInsetsForFrame:inSuperview:]):

Override this private helper method instead of `-safeAreaInsets`, since this is the ultimate source
of truth that determines `-safeAreaInsets`. This is consistent with `WebKitTestRunner`&apos;s
`TestRunnerWKWebView` subclass, which also has a mechanism for overriding safe area insets.

(-[TestWKWebView safeAreaInsets]): Deleted.

Canonical link: <a href="https://commits.webkit.org/264473@main">https://commits.webkit.org/264473@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3c16e246aa97a1c4744df19338c61de6a1b7c819

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7689 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7959 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8141 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9327 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7853 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7695 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9949 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7879 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10723 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7823 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8946 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7068 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9436 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6254 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/7000 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14672 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7417 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7122 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10533 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7613 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6211 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6943 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1846 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11154 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7347 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -199,7 +199,6 @@
 		31B76E4523299BDC007FED2C /* system-preview-trigger.html in Copy Resources */ = {isa = PBXBuildFile; fileRef = 31B76E4423299BA3007FED2C /* system-preview-trigger.html */; };
 		31E9BDA1247F4C62002E51A2 /* WebGLPrepareDisplayOnWebThread.mm in Sources */ = {isa = PBXBuildFile; fileRef = 31E9BDA0247F4C62002E51A2 /* WebGLPrepareDisplayOnWebThread.mm */; };
 		31E9BDA3247F5729002E51A2 /* webgl.html in Copy Resources */ = {isa = PBXBuildFile; fileRef = 31E9BDA2247F4DD0002E51A2 /* webgl.html */; };
-		3302A64E2A11B5350093FA2A /* DragAndDropSimulator.mm in Sources */ = {isa = PBXBuildFile; fileRef = 3302A64D2A11B4410093FA2A /* DragAndDropSimulator.mm */; };
```
Thanks for cleaning this up!

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewSnapshot.mm:
```diff
@@ -405,13 +406,14 @@ - (void)takeSnapshotWithWebView:(WKWebView *)webView configuration:(WKSnapshotCo
     NSInteger viewWidth = 800;
     NSInteger viewHeight = 600;
-    RetainPtr<WKWebView> webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, viewWidth, viewHeight)]);
+    // FIXME: This test fails when adopting TestWKWebView; it might be interesting to investigate why.
+    auto webView = adoptNS([[WKWebView alloc] initWithFrame:NSMakeRect(0, 0, viewWidth, viewHeight)]);
```
But it passes on both SE and other device simulators as-is? Interesting indeed

aprotyas reviewed (approved):
non-reviewer r+

whsieh committed:
[iOS] Adjust several API tests so that they pass when run against any iOS simulator
https://bugs.webkit.org/show_bug.cgi?id=257246
rdar://109743641

Reviewed by Aditya Keerthi.

Make some additional adjustments to several API tests, which currently pass in the iPhone SE
simulator but fail when run against other simulator models. These tests are:

• A subset of `WKWebView.SnapshotImage*`
• A subset of `AnimatedResize.*`
• `WebKitLegacy.RenderInContextSnapshot`

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Drive-by fix: address a (benign) build warning following 264080@main, where the definition of
`-runFromElement:toElement` is compiled twice, due to the fact that `DragAndDropSimulator.mm`
appears in _both_ the unified sources list as well as the Xcode project as a compilation target.
Address this by removing it as a compilation target in Xcode.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/AnimatedResize.mm:

Fix these tests by subclassing `TestWKWebView`, which normalizes safe area inset values between
different device models.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewSnapshot.mm:

Fix these tests by adopting `TestWKWebView` in place of `WKWebView`.

* Tools/TestWebKitAPI/Tests/WebKitLegacy/ios/SnapshotViaRenderInContext.mm:
(-[UIWebViewWithoutSafeArea _safeAreaInsetsForFrame:inSuperview:]):

Fix this test by creating a `UIWebView` subclass that simulates 0 safe area insets.

* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView _safeAreaInsetsForFrame:inSuperview:]):

Override this private helper method instead of `-safeAreaInsets`, since this is the ultimate source
of truth that determines `-safeAreaInsets`. This is consistent with `WebKitTestRunner`'s
`TestRunnerWKWebView` subclass, which also has a mechanism for overriding safe area insets.

(-[TestWKWebView safeAreaInsets]): Deleted.

Canonical link: https://commits.webkit.org/264473@main
